metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -1196,6 +1199,22 @@ def get_connection_object_page(
                 retry_on_auth_error=False,
                 retries_remaining=retries_remaining - 1,
             )
+            
+        except InternalServerError as ise:
+            # In some cases Tableau Server returns 504 error, which is a timeout error, so it worths to retry.
+            if ise.code == 504:
+                if retries_remaining <= 0:
+                    raise
+                return self.get_connection_object_page(
+                    query=query,
+                    connection_type=connection_type,
+                    query_filter=query_filter,
+                    fetch_size=fetch_size,
+                    current_cursor=current_cursor,
+                    retry_on_auth_error=False,
+                    retries_remaining=retries_remaining - 1,
```

Comment:
not a huge fan of this entire method - it feels way more complicated than it needs to be and has a ton of mostly duplicated code

Reply:
I agree
If this works, we should invest some time on a refactor.